code/game/objects/structures/girders.dm

```diff
@@ -294,6 +294,8 @@
 
 /obj/structure/girder/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	if(mover.pass_flags & PASSGIRDER)
+		return TRUE
 	if((mover.pass_flags & PASSGRILLE) || istype(mover, /obj/projectile))
```

Comment:
remove this check, add passgirder to anything that needs it, and dont give projectiles passgirder.

Reply:
this would change existing projectile behavior and balance, not something I want to get into on a meteors PR

Reply:
not if you do it correctly. the behavior would be the exact same you'd just fix the fact that girders use a flag that isn't about girders.

Reply:
can you explain in detail because i don't see how to do this without removing the existing 30% girder pass chance for projectiles and still allowing meteors to pass 100% of the time

Reply:
oh wait i get it now

Reply:
wait no that wouldn't work because then meteors would get stopped 60% of the time again

Reply:
bruh

1. add pass_girder to anything that had pass_grille EXCEPT for projectiles
2.remove the second check
3. done
4.

Reply:
but then projectiles can't go through girders at all

that is a balance change to projectiles

projectiles have a 30% chance to pass through girders as-is

Reply:
literally look a line below this

Reply:
i misread the second check, but in that case this is even worse, pass_girder shouldn't imply a CHANCE to pass, wtf.

Reply:
pass_girder doesn't imply a chance to pass
if pass_girder is there, it passes 100% of the time
this is literally how the code you are replying to works

Reply:
my bad i meant pass grill

Reply:
basically i just want you to remove PASS_GRILLE (you can leave the istype) because the new PASS_GIRDER implies that only that would let you pass girders, which is not the case in this code.

Reply:
shouldnt passgirder be in pass_flags_self?